src/time_zone_impl.cc

```diff
@@ -58,7 +61,7 @@ bool time_zone::Impl::LoadTimeZone(const std::string& name, time_zone* tz) {
 
   // Check whether the time zone has already been loaded.
   {
-    std::lock_guard<std::mutex> lock(TimeZoneMutex());
+    read_lock lock(TimeZoneMutex());
```

Comment:
If you are seeing contention on this mutex, which is only held over an `unordered_map` lookup, then I can only suggest that you are calling `cctz::load_time_zone()` way too much.  Typically you would only load a time zone once, and then use the returned `cctz::time_zone` value after that.  But if you need to build your own `string`->`cctz::time_zone` cache, that you can tune to your own needs, so be it.

Reply:
we made a cache in [OLAP]starrocks(https://github.com/StarRocks/starrocks/pull/2990), to avoid calling cctz inner functions too much times, at the same time, we also want to optimize lock in cctz.

Reply:
about the lock of optimize, https://github.com/StarRocks/starrocks/pull/2966

Reply:
I'm sure it is easier for you to build a high-performance cache given the platform and application knowledge of your app than it is for me in a general-purpose library using only C++11.  So, good luck.

[Aside:  I would wonder about the model behind anything that uses the term "timezone_offsets".]